config/routes.rb

```diff
@@ -1,6 +1,11 @@
 Rails.application.routes.draw do
+  devise_for :users
   root 'users#index'
-  resources :users, only: [:index, :show]  do 
-    resources :posts, only: [:index, :show] 
+
+  resources :users, only: [:index,  :show] do
+    resources :posts, only: [:index,  :show, :new, :create, :destroy] do 
+      resources :comments, only: [ :new, :index, :create, :destroy]
+      resources :likes, only: [:create]
+    end
   end
```

Comment:
- [ ] You used the default actions in rails which responds to .json format as we requested, but you haven't set any API endpoints as per the requirement. It's a common practice to write the actions on your own for learning purposes and create an API folder with controllers just returning json and you could set them in this routes.rb file like: '/api/users/user_id/posts' or 'api/v1/users/:user_id/posts' for example to make a GET request. Kindly create all endpoints under these routes for an API and don't forget to write the necessary methods(index, create..) which will only get json and return json objects:
![image](https://github.com/Ridwanullahi-code/BlogApp/assets/91486791/8ecc4c43-7680-4f42-bfd5-6a1b619126c2)